core/src/test/java/org/elasticsearch/search/suggest/AbstractSuggestionBuilderTestCase.java

```diff
@@ -131,6 +151,69 @@ public void testFromXContent() throws IOException {
         }
     }
 
+    public void testBuild() throws IOException {
+        Settings indexSettings = Settings.builder().put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT).build();
+        IndexSettings idxSettings = IndexSettingsModule.newIndexSettings(new Index(randomAlphaOfLengthBetween(1, 10), "_na_"),
+                indexSettings);
+        MapperService mapperService = mock(MapperService.class);
+        ScriptService scriptService = mock(ScriptService.class);
+        MappedFieldType fieldType = mockFieldType();
+        boolean fieldTypeSearchAnalyzerSet = randomBoolean();
+        if (fieldTypeSearchAnalyzerSet) {
+            NamedAnalyzer searchAnalyzer = new NamedAnalyzer("fieldSearchAnalyzer", AnalyzerScope.INDEX, new SimpleAnalyzer());
+            if (Mockito.mockingDetails(fieldType).isMock()) {
+                when(fieldType.searchAnalyzer()).thenReturn(searchAnalyzer);
+            } else {
+                fieldType.setSearchAnalyzer(searchAnalyzer);
+            }
+        } else {
+            when(mapperService.searchAnalyzer())
+                .thenReturn(new NamedAnalyzer("mapperServiceSearchAnalyzer", AnalyzerScope.INDEX, new SimpleAnalyzer()));
+        }
+        when(mapperService.fullName(any(String.class))).thenReturn(fieldType);
+        when(mapperService.getNamedAnalyzer(any(String.class))).then(
+                invocation -> new NamedAnalyzer((String) invocation.getArguments()[0], AnalyzerScope.INDEX, new SimpleAnalyzer()));
+        when(scriptService.compile(any(Script.class), any())).then(
+                invocation -> new TestTemplateService.MockTemplateScript.Factory(((Script) invocation.getArguments()[0]).getIdOrCode()));
```

Comment:
Use the original Script code for MockTemplateScript, this will be checked in https://github.com/elastic/elasticsearch/pull/25571/files#diff-7925b951f7645dcb8a552ff8440accd3R208